kyo-core/shared/src/main/scala/kyo/scheduler/IOPromise.scala

```diff
@@ -63,12 +63,13 @@ private[kyo] class IOPromise[E, A](init: State[E, A]) extends Safepoint.Intercep
     end interrupts
 
     final def mask: IOPromise[E, A] =
-        val p = IOPromise[E, A]()
+        val p = new IOPromise[E, A]:
+            override def interrupt(error: Panic): Boolean = false
```

Comment:
I had avoided implementing this behavior before because I thought we'd need to add a new type to `IOPromise.State` but, given that `interrupt` is the only method that exposes interruption, we can simply override it here.